spec/type_encrypted_spec.rb

```diff
@@ -0,0 +1,100 @@
+require File.expand_path("../support", __FILE__)
+
+require "active_model"
+
+class SubTypeEncryptedTest < CouchbaseOrm::NestedDocument
+    attribute :name, :string
+    attribute :secret, :encrypted
+end
+
+class TypeEncryptedTest < CouchbaseOrm::Base
+    attribute :main, :nested, type: SubTypeEncryptedTest
+    attribute :others, :array, type: SubTypeEncryptedTest
+    attribute :secret, :encrypted
+end
+
+class SpecificAlgoTest < CouchbaseOrm::Base
+    attribute :secret, :encrypted, alg: "3DES"
+end
+
+describe CouchbaseOrm::Types::Encrypted do
+    it "prefix attribute on serialization" do
+        obj = TypeEncryptedTest.new(secret: "mysecret")
+        expect_serialized_secret(obj)
+    end
+
+    it "prefix attribute on nested objects" do
+        obj = TypeEncryptedTest.new(main: SubTypeEncryptedTest.new(secret: "mysecret"))
+        expect_serialized_secret(obj.main)
+    end
+
+    it "prefix attribute on array objects" do
+        obj = TypeEncryptedTest.new(others: [SubTypeEncryptedTest.new(secret: "mysecret")])
+        expect_serialized_secret(obj.others.first)
+    end
+
+    def expect_serialized_secret(obj)
+        expect(obj.send(:serialized_attributes)["encrypted$secret"]).to eq({alg:"CB_MOBILE_CUSTOM", ciphertext: "mysecret"})
+        expect(obj.send(:serialized_attributes)).to_not include "secret"
```

Comment:
I had to check the doc of [Hash#include?](https://ruby-doc.org/core-3.1.2/Hash.html#method-i-include-3F) find `has_key?` alias more clear.

```suggestion
        expect(obj.send(:serialized_attributes)).to_not have_key "secret"
```